CRM/Sqltasks/Task.php

```diff
@@ -188,15 +189,46 @@ public function setConfiguration($config, $writeTrough = FALSE) {
    * Append log messages
    *
    * @param $message
+   * @param null $type
+   * @param bool $skipRegularLog
    */
-  public function log($message) {
+  public function log($message, $type = NULL, $skipRegularLog = FALSE) {
```

Comment:
I like the introduction of `$type` - we should use this to e.g. render error-level messages in red when displaying the log. We'll also need to add this in a lot of places where we generate logs.